src/index.js

```diff
@@ -1447,5 +1452,266 @@ module.exports = {
+            if (childType.is('DOMElement')) {
+              // Element
+              score += scoreElementAgainstSpec(
+                element.childNodes[i],
+                convertDOMNodeToSatisfySpec(childSpec)
+              );
+            }
+
+            score++;
+          } else if (expect.findTypeOf(childSpec).is('DOMIgnoreComment')) {
+            score++;
+          }
+        } else if (
+          childType.is('DOMElement') &&
+          typeof childSpec === 'object'
+        ) {
+          score += scoreElementAgainstSpec(element.childNodes[i], childSpec);
+        } else if (
+          childType.is('DOMTextNode') &&
+          isTextMatching(child.nodeValue, childSpec)
+        ) {
+          score++;
+        }
+      });
+
+      return score;
+    }
+
+    function findMatchesWithGoodScore(data, spec) {
+      const elements =
+        typeof data.length === 'number' ? Array.from(data) : [data];
+
+      const result = [];
+      let bestScore = 0;
+
+      elements.forEach(element => {
+        const score = scoreElementAgainstSpec(element, spec);
+        bestScore = Math.max(score, bestScore);
+
+        if (score > 0 && score >= bestScore) {
+          result.push({ score, element });
+        }
+
+        for (var i = 0; i < element.childNodes.length; i += 1) {
+          const child = element.childNodes[i];
+          if (child.nodeType === 1) {
+            result.push(...findMatchesWithGoodScore(child, spec));
+          }
+        }
+      });
+
+      result.sort((a, b) => b.score - a.score);
+
+      if (result.length > 0) {
+        const bestScore = result[0].score;
+
+        return result.filter(({ score }) => score === bestScore);
+      }
+
+      return result;
+    }
+
+    expect.exportAssertion(
+      '<DOMDocument|DOMElement|DOMDocumentFragment|DOMNodeList> [not] to contain <DOMElement|object|string>',
+      (expect, subject, value) => {
+        const nodes = subject.childNodes || makeAttachedDOMNodeList(subject);
+        const isHtml = isInsideHtmlDocument(
+          subject.childNodes ? subject : nodes
+        );
+        const valueType = expect.findTypeOf(value);
+        let spec = value;
+
+        if (valueType.is('DOMElement')) {
+          spec = convertDOMNodeToSatisfySpec(value, isHtml);
+        } else if (valueType.is('string')) {
+          const documentFragment = isHtml
+            ? parseHtml(value, true)
+            : parseXml(value);
+
+          if (documentFragment.childNodes.length !== 1) {
+            throw new Error(
+              'HTMLElement to contain string: Only a single node is supported'
+            );
+          }
+
+          spec = convertDOMNodeToSatisfySpec(
+            documentFragment.childNodes[0],
+            isHtml
+          );
+
+          if (typeof spec === 'string') {
+            throw new Error(
+              'HTMLElement to contain string: please provide a HTML structure as a string'
```

Comment:
Just for full disclosure -- IMO it'd be fine to support a RHS of a string without HTML tags wrapped around it -- as long as `<` etc. are interpreted as HTML rather than text.

We can also add that later, so nbd.

Reply:
@papandreou I'm leaning towards having an explicit 
to contain text assertion, that also mirror the current to have text and to satisfy assertions better. What du you think about that idea?

Reply:
That could be a fine addition of its own. I still think `<DOMDocument|DOMElement|...> to contain <string>` should behave as I suggest above.

Reply:
But I think that is what we had, but I saw some weirdness around HTML entities so I just reverted it. Let's see if we can reintroduce it in a later PR.